java/java-analysis-impl/src/com/intellij/codeInspection/dataFlow/HardcodedContracts.java

```diff
@@ -319,6 +350,42 @@ static List<MethodContract> mathMinMax(boolean isMax) {
                          trivialContract(returnParameter(1)));
   }
 
+  private static List<MethodContract> javaTimeCompareWithCheckType(RelationType relationType, @Nullable SpecialField specialField) {
+    ContractValue argument = ContractValue.argument(0);
+    ContractValue qualifier = ContractValue.qualifier();
+
+    if (specialField != null) {
+      qualifier = qualifier.specialField(specialField);
+      argument = argument.specialField(specialField);
+    }
+
+    return Arrays.asList(
+      multipleConditionContract(
+        List.of(
+          ContractValue.condition(qualifier, relationType, argument),
+          ContractValue.condition(argument, RelationType.IS, qualifier)),
```

Comment:
What relation do you want to express with 'IS'? This part is unclear to me. What will happen if this clause is omitted?